pkg/manager/member/tidb_member_manager.go

```diff
@@ -245,6 +255,103 @@ func (m *tidbMemberManager) syncTiDBStatefulSetForTidbCluster(tc *v1alpha1.TidbC
 	return mngerutils.UpdateStatefulSetWithPrecheck(m.deps, tc, "FailedUpdateTiDBSTS", newTiDBSet, oldTiDBSet)
 }
 
+func (m *tidbMemberManager) syncInitializer(tc *v1alpha1.TidbCluster) {
+	// set random password
+	ns := tc.Namespace
+	tcName := tc.Name
+	//check endpoints ready
+	isTiDBReady := false
+	eps, epErr := m.deps.EndpointLister.Endpoints(ns).Get(controller.TiDBMemberName(tcName))
+	if epErr != nil {
+		klog.Errorf("Failed to get endpoints %s for cluster %s/%s, err: %s", controller.TiDBMemberName(tcName), ns, tcName, epErr)
+		return
+	}
+	// TiDB service has endpoints
+	if eps != nil && len(eps.Subsets[0].Addresses) > 0 {
+		isTiDBReady = true
+	}
+
+	if !isTiDBReady {
+		klog.Infof("Wait for TiDB ready for cluster %s/%s", ns, tcName)
+		return
+	}
+	// sync password secret
+	var password string
+	secretName := controller.TiDBInitSecret(tcName)
+	secret, err := m.deps.SecretLister.Secrets(ns).Get(secretName)
+	passwordSecretExist := true
+	if err != nil {
+		if errors.IsNotFound(err) {
+			passwordSecretExist = false
+		} else {
+			klog.Errorf("Failed to get secret %s for cluster %s/%s, err: %s", secretName, ns, tcName, epErr)
+			return
+		}
+	}
+
+	if !passwordSecretExist {
+		klog.Infof("Create random password secret for cluster %s/%s", ns, tcName)
+		var secret *corev1.Secret
+		secret, password = m.buildRandomPasswordSecret(tc)
+		err := m.deps.TypedControl.Create(tc, secret)
+		if err != nil {
+			klog.Errorf("Failed to create secret %s for cluster %s:%s, err: %s", secretName, ns, tcName, err)
+			return
+		}
+	} else {
+		password = string(secret.Data[constants.TidbRootKey])
+	}
+	// init password
+	var db *sql.DB
+	dsn := util.GetDSN(tc)
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+	db, err = util.OpenDB(ctx, dsn)
+
+	if err != nil {
+		if ctx.Err() != nil {
+			klog.Errorf("Can't connect to the TiDB service of the TiDB cluster [%s:%s], error: %s, context error: %s", ns, tcName, err, ctx.Err())
+		} else {
+			klog.Errorf("Can't connect to the TiDB service of the TiDB cluster [%s:%s], error: %s", ns, tcName, err)
+		}
+		return
+	} else {
+		klog.Infof("Set random password for cluster %s/%s", ns, tcName)
+		defer func(db *sql.DB) {
+			err := db.Close()
+			if err != nil {
+				klog.Errorf("Closed db connection for TiDB cluster %s/%s, err: %v", ns, tcName, err)
+			}
+		}(db)
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+		err = util.SetPassword(ctx, db, password)
+		if err != nil {
+			klog.Errorf("Fail to set TiDB password for TiDB cluster %s/%s, err: %s", ns, tcName, err)
+			return
+		}
+		tc.Status.TiDB.PasswordInitialized = true
```

Comment:
do we need to handle `SetPassword` successfully, but update tc stats failed?

Reply:
How should I handle this situation?

Reply:
I have add Access denied error check , if error is auth error, i will skip this step.